src/mip/mip_parser.h

```diff
@@ -61,6 +61,15 @@ typedef struct mip_parser
     byte_ring_state     _ring;                                 ///<@private Ring buffer which holds data being parsed. User-specified backing buffer and size.
     mip_packet_callback _callback;                             ///<@private Callback called when a valid packet is parsed. Can be NULL.
     void*               _callback_object;                      ///<@private User-specified pointer passed to the callback function.
+
+#ifdef MIP_ENABLE_DIAGNOSTIC_COUNTERS
+    uint32_t            _diag_bytes_parsed;                    ///<@private Counts the total number of bytes read from the user input buffer.
```

Comment:
Would it make sense to put this stuff in a seperate struct that gets put in this struct so there could be a header something like `mip_diagnostics.h` that has all the diagnostics definitions in it?

Reply:
I was thinking moving to it's own header would be convenient so that all the `#ifdef`s and macro stuff in one header file so that the `mip_parser.h` could be free of all that. Not required, and maybe there would still need to be some `#ifdef` stuff in the parser anyway

Reply:
There are only two #ifdefs for the diagnostic definitions in the header and two in the source file. We could hide the struct definitions behind a macro or something, but IMHO that would just obfuscate things.

Defining a MIP_DIAG_ZERO macro is a good idea though, and would get rid of one #ifdef.